quake1_example/class/_func.gd

```diff
@@ -1,14 +1,61 @@
 extends AnimatableBody3D
 class_name QmapbspQuakeFunctionBrush
 
+# Spawnflags:
+# 256 = Not on Easy
+# 512 = Not on Normal
+# 1024 = Not on Hard
+# 2048 = Not in Deathmatch
+const NOT_IN_EASY : int = 256
+const NOT_IN_NORMAL : int = 512
+const NOT_IN_HARD : int = 1024
+const NOT_IN_DM : int = 2048
+
 var props : Dictionary
+var aabb : AABB
+
+
 func _get_properties(dict : Dictionary) : props = dict
 
-var aabb : AABB
+
+func _flag(spawnflag : int) -> bool :
+	if not props.has(&'spawnflags') : 
+		printerr('Missing Spawnflags! ', spawnflag, ' ', self)
+		return false
+	var result : bool = _prop(&'spawnflags', 0) & spawnflag
+	return result
+
+
+func _prop(name : StringName, def) :
+	var result = props.get(name, def)
+	if typeof(result) == typeof(def) : return result
+	match typeof(def) :
+		TYPE_NIL:
+			result = def
+		TYPE_BOOL:
+			result = result.to_bool()
+		TYPE_INT:
+			result = result.to_int()
+		TYPE_FLOAT:
+			result = result.to_float()
+		TYPE_STRING:
+			result = str(result)
+		_ :
+			result = def
+	return result
```

Comment:
I've seen this one in `_trigger`. Would it be better if we move this function to be a static function in a util class which any class can be accessed ?

Reply:
I recommended using regular strings as a dictionary key. Since the StringNames and strings are considered the same. So the GDScript wouldn't take time to construct a new StringName for a key. (Constructing a new StringName is slightly slower than a regular string. except the case if the StringName is a literal form like `&"abc"`)

Reply:
Aren't StringNames better for repetitive keys, so they point to the same memory space instead of alocating new memory for all key names? and for the second part `... case if the StringName is a literal form like &"abc")` -- isn't that the case?

Reply:
Godot dictionary may not work as you think. Try running this snippet.
```gdscript
var dict := {}
dict[&'a'] = 1
print(dict.find_key(1) is StringName)
```
This will print `false`.

Godot will internally convert the StringName keys to strings. I think in practice, StringName and string have the same content and their hash should be the same too. Sometimes I argue against this behavior too. It might confuse end users.

But originally, the StringNames were exposed to GDScript in 4.0 for exchanging some engine properties that have unique names e.g. node names, method names, and signal names.

Reply:
Seems it's not like ruby's hash/dictionary then, missed opportunity IMHO -- I'll change dict keys to strings then :)